### PR TITLE
Introduce the commands for API add, delete, update and generate deployment directory for Kubernetes

### PR DESCRIPTION
**Pull request by Jayanie**
## Purpose

- This PR introduces the commands for API add, delete, update and generate deployment directory for Kubernetes.
- This PR is related to:
     1. https://github.com/wso2/k8s-api-operator/pull/561
     2. https://github.com/wso2/k8s-api-operator/pull/503
     3. https://github.com/wso2/product-apim-tooling/pull/622

### CODE REVIEW

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Managing APIs in Kubernetes
+
+WSO2 API Controller, **apictl** allows you to manage APIs in Kubernetes with the use of WSO2 API operator. This approach uses the kubernetes native way to deploy the APIS into a running microgateway cluster from the command line.
```
Comment (wasuradananjith) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:3]:
```suggestion
**WSO2 API Controller (apictl)** allows you to manage APIs in Kubernetes with the use of WSO2 API Operator. This approach uses the Kubernetes native way to deploy the APIS into a running microgateway cluster from the command line.
```
Recently we decided to introduce **apictl** abbreviation as shown above when starting any apictl related doc. After that, you can use this abbreviation through the doc. We can do the same for API Operator as well if there is an abbreviation for that.

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    - [Kubernetes v1.14 or above](https://Kubernetes.io/docs/setup/) <br>
+
+        - Minimum CPU : 2vCPU
+        - Minimum Memory : 2GB
+
+!!! info
+    **Before you begin** 
+
+    -  First download the microgateway distribution from the [github release page](https://github.com/wso2/product-microgateway/releases/) and extract that to a folder of your choice. Then deploy microgateway in kubernetes cluster. Please refer [Steps for deploying  microgateway in kubernetes]()
```
Comment (Chamindu36) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:17]: Don't we need to add the docs link after this for the steps for the mentioned deployment?
**[Steps for deploying microgateway in kubernetes]**

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    -  Make sure the CTL is downloaded and initialized, if not follow the steps in [Download and Initialize the CTL Tool]({{base_path}}/install-and-setup/setup/api-controller/getting-started-with-wso2-api-controller/#download-and-initialize-the-ctl-tool).
```
Comment (wasuradananjith) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:19]:
```suggestion
    -  Make sure the apictl is downloaded and initialized, if not follow the steps in [Download and initialize the apictl]({{base_path}}/install-and-setup/setup/api-controller/getting-started-with-wso2-api-controller/#download-and-initialize-the-apictl).
```
Please recheck the links. These have been changed recently.

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+    -  Ensure that the API Operator is deployed in the kubernetes cluster. Please refer [API Operator documentation](https://github.com/wso2/k8s-api-operator/tree/master) for detailed steps.
```
Comment (wasuradananjith) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:21]:
```suggestion
    -  Ensure that the API Operator is deployed in the Kubernetes cluster. Please refer [API Operator documentation](https://github.com/wso2/k8s-api-operator/tree/master) for detailed steps.
```

Comment (wasuradananjith) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:21]: Let's make the first letter uppercase here **Kubernetes** ?

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/resources/kustomize-config.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  - path: spec/swaggerConfigMapName
+    kind: API
+- kind: ConfigMap
+  fieldSpecs:
+  - path: spec/paramsValues
+    kind: API
+- kind: Secret
+  fieldSpecs:
+  - path: spec/certsValues
+    kind: API
```
Comment (wasuradananjith) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/resources/kustomize-config.yaml:13]: Shall we add a newline at the EOF?

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+            ```go
+             apictl k8s gen deployment-dir -s PizzaShackAPI-1.0.0 -d Deployment-Directory
+            ```
+
+    -   **Response**
+
+        ```go
+        The deployment directory for PizzaShackAPI-1.0.0 file is generated at Deployment-Directory
+        ```
+
```
Comment (wasuradananjith) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:156]: Shall we mention what will be generated? The folder structure perhaps as we mentioned in the apictl - **Configuring Environment Specific Parameters** page.
image

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    -  First download the microgateway distribution from the [github release page](https://github.com/wso2/product-microgateway/releases/) and extract that to a folder of your choice. Then deploy microgateway in kubernetes cluster. Please refer [Steps for deploying  microgateway in kubernetes]()
+
+    -  Make sure the CTL is downloaded and initialized, if not follow the steps in [Download and Initialize the CTL Tool]({{base_path}}/install-and-setup/setup/api-controller/getting-started-with-wso2-api-controller/#download-and-initialize-the-ctl-tool).
+
+    -  Ensure that the API Operator is deployed in the kubernetes cluster. Please refer [API Operator documentation](https://github.com/wso2/k8s-api-operator/tree/master) for detailed steps.
+
+## Add an API to Kubernetes cluster
+
+1.  Run the following command to add an API to the Kubernetes cluster using either a swagger file, zip file, or an API project folder. When you add the API to the Kubernetes cluster, the API operator successfully deploys the project zip to the WSO2 microgateway Adapter component.
```
Comment (wasuradananjith) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:25]:
```suggestion
1.  Run the following command to add an API to the Kubernetes cluster using either a swagger file, zip file, or an API project folder. When you add the API to the Kubernetes cluster, the API operator successfully deploys the project zip to the WSO2 Microgateway Adapter component.
```
Let's make the first letter uppercase **Microgateway**?

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/resources/kustomization.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+ - PizzaShackAPI-params.yaml
+
+# kustomized-used: true
+
+#kubect label change kustomized-used: false
+
+#kubectl kustomize over/ | kubectl apply -f 
+
+#kubectl delete cm -l kustomized-used: false
+
```
Comment (Chamindu36) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/resources/kustomization.yaml:41]: Shall we remove the unwanted newlines here?

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+3.  Run the following command to generate the related config maps.
+
+    -   **Command** 
+
+        ```go
+        kubectl kustomize
+
+        ```
+
+
```
Comment (Chamindu36) [inline en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:168]: Shall we remove the extra new line here?

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md (hunk 2 above)
Comment (Chamindu36) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:17]: And Shall we change the reference subject as  "Steps for deploying  Microgateway in Kubernetes" as @wasuradananjith mentioned.

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md (hunk 1 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md]: Fixed with 21f466ce5d99225a24a45351eae0997e87f8c2f5

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md (hunk 4 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md]: Fixes with 21f466ce5d99225a24a45351eae0997e87f8c2f5

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md (hunk 7 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:27]: Fixed with 21f466ce5d99225a24a45351eae0997e87f8c2f5

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md (hunk 9 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md]: Fixed with 21f466ce5d99225a24a45351eae0997e87f8c2f5

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/resources/kustomization.yaml (hunk 8 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/resources/kustomization.yaml]: Fixed with 21f466ce5d99225a24a45351eae0997e87f8c2f5

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/resources/kustomize-config.yaml (hunk 5 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/resources/kustomize-config.yaml]: Fixed with 21f466ce5d99225a24a45351eae0997e87f8c2f5

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md (hunk 2 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md]: Fixed with 21f466ce5d99225a24a45351eae0997e87f8c2f5

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md (hunk 3 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md]: Fixed with 21f466ce5d99225a24a45351eae0997e87f8c2f5

File: en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md (hunk 6 above)
Comment (Jayanie) [reply en/docs/install-and-setup/setup/api-controller/managing-apis-in-kubernetes/managing-apis-in-kubernetes.md:158]: Fixed with 3cbe17367ae5223e044356397e754ce547859314

**Review by wasuradananjith — APPROVED** [review]

**Review by 1akshitha — APPROVED** [review]

**Review by Chamindu36 — APPROVED** [review]

**Pull request merged**